celo feedback

RaphaelNdonga opened the pull request.

RaphaelNdonga committed:
working version of nftlottery contract

RaphaelNdonga committed:
buy tickets issue fixed

RaphaelNdonga committed:
no of tickets returned to dynamic

RaphaelNdonga committed:
buyTicket doesn't have to be payable

RaphaelNdonga committed:
startLottery.js file created using hardhat

RaphaelNdonga committed:
created utils file to contain utility functions for lottery address

RaphaelNdonga committed:
getWinning ticket method created using hardhat

RaphaelNdonga committed:
payoutwinner changed to use hardhat

JoE11-y reviewed (changes requested):
Please add a description outlining all the changes you made resolve the issue.

I saw the hardhat implementation for the operator functions.

RaphaelNdonga commented:
Hi Joel. I explained the changes in the dacade platform. Let me post it here:

Hello JoeEdoh. Good attempt on your simple lottery contract. However, there was one major problem. Your buyTicket function does not function properly. I figured the reason was because you had set the time interval for your lottery was 2 hours. I changed this to 2 weeks.

I also made the following changes:

setting the operator.
I modified the constructor to set the operator to the creator of the contract when the contract is initially deployed. This avoids the redundancy of having to create a setOperator function under 'operator' folder. It does the same thing of setting the operator to the creator of the contract.

buy ticket does not have to be payable.
This is a common mistake because ERC20 tokens are used for paying. However, payable functions should only be used when the native currency is being exchanged. In this case, if you were using Celo instead of cUSD for buying tickets, only then, should you make the function payable.

operator folder.
This folder contains all your operations on the NFTLottery. It may function properly, but it is not advisable to use this approach. It is laborious, setting up the contract kits and method variables. A better way would be to use Hardhat to perform operations on the NFTLottery. I have demonstrated this by replicating some of the operator functions in the script folder. I have left out a few. You can challenge yourself by implementing them.

All in all, good work. Keep building.

JoE11-y reviewed (approved):
Okay this works, thanks

JoE11-y committed:
Merge branch 'main' into main
